src/net_processing.cpp

```diff
@@ -271,12 +271,15 @@ struct Peer {
 
     struct TxRelay {
         mutable RecursiveMutex m_bloom_filter_mutex;
-        /** Whether the peer wishes to receive transaction announcements.
+        /** Whether we relay transactions to this peer.
          *
-         * This is initially set based on the fRelay flag in the received
-         * `version` message. If initially set to false, it can only be flipped
-         * to true if we have offered the peer NODE_BLOOM services and it sends
-         * us a `filterload` or `filterclear` message. See BIP37. */
+         * On receiving a `version` message, this is initially set to false
+         * if the peer is outbound block-relay-only and otherwise based on
+         * the fRelay flag.
```

Comment:
I think that if this is an outbound block-relay-only peer, we don't create a `TxRelay` object for this peer in the first place, so  `TxRelay::m_relay_txs` is neither true nor false but doesn't exist.

Reply:
Thanks, updated. I agree this was previously out of date and the proposed update was imprecise. It is already documented in the `PeerManagerImpl::ProcessMessage()::VERSION` code so it seems best to remove it here instead of duplicating.